### PR TITLE
get_status() RFC Implementation

### PR DESCRIPTION
**Pull request by Sami1309**
# Description

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: client/src/featureform/register.py [hunk 1]
```diff
@@ -2572,6 +2572,50 @@ def apply(self):
         else:
             state().create_all(self._stub)
 
+    def wait(self, resource_type, name, variant=None, timeout=None):
+        # gets a resource and waits until status is set to ready
+        resource_functions = {
```
Comment (sdreyer) [inline client/src/featureform/register.py:2577]: Do we need these?

File: client/src/featureform/register.py [hunk 2]
```diff
@@ -2572,6 +2572,50 @@ def apply(self):
...
+        resource_variant_functions = {
+            "feature": self.get_feature,
+            "label": self.get_label,
+            "source": self.get_source,
+            "trainingset": self.get_training_set,
+            "training-set": self.get_training_set,
+        }
+        
+        resource = None
+        if variant is None:
```
Comment (sdreyer) [inline client/src/featureform/register.py:2593]: What are we waiting for if theres no variant?

File: client/src/featureform/register.py [hunk 3]
```diff
@@ -2572,6 +2572,50 @@ def apply(self):
...
+            resource = resource_functions[resource_type](name, local=self.local)
+        else:
+            resource = resource_variant_functions[resource_type](name, variant, local=self.local)
+        status = resource.get_status()
+        
+        timeout_duration = timedelta(seconds=0)
```
Comment (sdreyer) [inline client/src/featureform/register.py:2599]: These timeouts could probably be returned from a function

File: client/src/featureform/register.py [hunk 4]
```diff
@@ -2572,6 +2572,50 @@ def apply(self):
...
+        if timeout is not None:
+            timeout_duration = timedelta(seconds=timeout)
+        time_waited = timedelta(seconds = 0)
+        time_started = datetime.now()
+
+        while (status != ResourceStatus.Failed and status != ResourceStatus.Ready) and (timeout is None or time_waited < timeout_duration):
+            if variant is None:
+                resource = resource_functions[resource_type](name, local=self.local)
```
Comment (sdreyer) [inline client/src/featureform/register.py:2607]: What is this checking?

File: client/src/featureform/register.py [hunk 5]
```diff
@@ -2572,6 +2572,50 @@ def apply(self):
...
+            else:
+                resource = resource_variant_functions[resource_type](name, variant, local=self.local)
+            status = resource.get_status()
+            time.sleep(1)
```
Comment (sdreyer) [inline client/src/featureform/register.py:2611]: Why sleep after getting the status?

File: client/src/featureform/register.py (hunk 1 above)
Comment (Sami1309) [reply client/src/featureform/register.py:2577]: no, will remove them

File: client/src/featureform/register.py (hunk 2 above)
Comment (Sami1309) [reply client/src/featureform/register.py:2593]: mistakenly added

File: client/src/featureform/register.py (hunk 3 above)
Comment (Sami1309) [reply client/src/featureform/register.py:2599]: what do you mean?

File: client/src/featureform/register.py (hunk 5 above)
Comment (Sami1309) [reply client/src/featureform/register.py:2611]: so the check doesn't run to frequently

File: client/src/featureform/register.py (hunk 4 above)
Comment (Sami1309) [reply client/src/featureform/register.py:2607]: checking if resource is no longer pending or created

**Pull request closed without merging**